src/providers/deno.rs

```diff
@@ -0,0 +1,37 @@
+use super::Provider;
+use crate::{nixpacks::app::App, providers::Pkg};
+use anyhow::Result;
+use regex::Regex;
+
+pub struct DenoProvider {}
+
+impl Provider for DenoProvider {
+    fn name(&self) -> &str {
+        "deno"
+    }
+
+    fn detect(&self, app: &App) -> Result<bool> {
+        if !app.includes_file("src/index.ts") {
+            false;
+        }
```

Comment:
This line actually does nothing, so we can remove it.

```suggestion
```

Reply:
It's because the run command references `src/index.ts` so I wanted to ensure it existed. We should probably make this. It should probably, instead, try to find either `index.ts` or `src/index.ts` and error if neither exist?

Reply:
Is index.ts / index.js *always* the entry point, or can this be defined somewhere? Is the src directory needed as well?

Reply:
Oh, actually it should just make sure that in the project there is either a JS or TS file. That can be done with glob, or perhaps introducing another function, `includes_glob`.

Reply:
Ya, will try something like that. Basically, run the first file that matches `**/index.{ts,js}`

Reply:
I think the problem is that Deno projects don't have a `package.json` file to inspect. For situations like this I don't think it is bad to go with a sane default. Sure it may not _always_ be the case, but it is likely. And users will be able to override the start command if they want. If this needs to be refined in the future then it can be. But one of the goals with this project is sane defaults with easy customization.

With that said, we could also check for `./index.ts` (no src) too since that is also pretty likely.

Reply:
> Ya, will try something like that. Basically, run the first file that matches `**/index.{ts,js}`

Oh I missed this in my reply. This is good I think

Reply:
Actually, how does the heroku buildpack detect it?